tests/integration/devfile/utils/utils.go

```diff
@@ -186,3 +189,81 @@ func ExecWithRestartAttribute(projectDirPath, cmpName, namespace string) {
 	Expect(output).To(ContainSubstring("if not running"))
 
 }
+
+// OdoWatch creates files, dir in the context and watches for the changes to be pushed
+// this was taken from e2e_images_test.go and modified for devfile
+func OdoWatch(cmpName, project, context string, stringsToBeMatched []string, runner interface{}, platform string) {
```

Comment:
How about using the OdoWatch function from the e2e_images_test.go itself by doing some modification over there. Like you can pass dynamic args to the function OdoWatch and include some checks if it is related to devfile.
As I can see you are replacing the app.js I think that can be handled smartly and yes you need to do modifications to some few places.

But this will avoid duplicacy of the code and ofcourse its maintainance also.

Reply:
done!